rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala

```diff
@@ -340,3 +340,140 @@ object MessageStatusReport {
+        ~ ( "expectedConfigId" -> e.nodeConfigId.value)
+        )
+      case UnexpectedVersion(t, id, _, e, _) =>
+        ( ( "type"             -> "UnexpectedVersion"       )
+        ~ ( "expectedConfigId" -> e.nodeConfigId.value      )
+        ~ ( "runConfigId"      -> id.get.nodeConfigId.value )
+        )
+      case UnexpectedNoVersion(_, id, _, e, _) =>
+        ( ( "type"             -> "UnexpectedNoVersion" )
+        ~ ( "expectedConfigId" -> e.nodeConfigId.value  )
+        ~ ( "runConfigId"      -> id.value              )
+        )
+      case UnexpectedUnknowVersion(_, id, e, _) =>
+        ( ( "type"             -> "UnexpectedUnknownVersion" )
+        ~ ( "expectedConfigId" -> e.nodeConfigId.value       )
+        ~ ( "runConfigId"      -> id.value                   )
+        )
+      case Pending(e, r, _) =>
+        ( ( "type"             -> "Pending"               )
+        ~ ( "expectedConfigId" -> e.nodeConfigId.value    )
+        ~ ("runConfigId"       -> r.map( _._2.nodeConfigId.value ) )
+        )
+      case ComputeCompliance(_, e, _) =>
+        ( ( "type"             -> "ComputeCompliance" )
+        ~ ( "expectedConfigId" -> e.nodeConfigId.value       )
+        ~ ( "runConfigId"      -> e.nodeConfigId.value       )
+        )
+    }
+  }
+  def jsonStatusInfo(statusInfo: RunComplianceInfo): JValue = {
+    (
+      ( "status" -> (statusInfo match {
+                      case RunComplianceInfo.OK => "success"
+                      case _                    => "error"
+                    })
+      ) ~ (
+        "errors" -> (statusInfo match {
+                      case RunComplianceInfo.OK => None
+                      case RunComplianceInfo.PolicyModeInconsistency(errors) => Some(errors.map {
+                        case RunComplianceInfo.PolicyModeError.TechniqueMixedMode(msg) =>
+                          ( "policyModeError" -> ( "message" -> msg ) ):JObject
+                        case RunComplianceInfo.PolicyModeError.AgentAbortMessage(cause, msg) =>
+                          ( "policyModeInconsistency" -> ( ( "cause" -> cause ) ~ ( "message" -> msg ) ) ):JObject
+                      })
+                    })
+      )
+    )
+  }
+
+  implicit class RunComplianceInfoToJs(x: (RunAndConfigInfo, RunComplianceInfo)) {
+    def toJValue() = {
+      (
+        ( "run"    -> jsonRunInfo(x._1)    )
+      ~ ( "status" -> jsonStatusInfo(x._2) )
+      )
+    }
+
+    def toJson() = pretty(render(toJValue))
+    def toCompactJson = compactRender(toJValue)
+  }
+
+  implicit class AggregatedStatusReportToJs(x: AggregatedStatusReport) {
+    import ComplianceLevelSerialisation._
+
+    def toJValue(): JValue = {
+
+      //here, I'm not sure that we want compliance or
+      //compliance percents. Having a normalized value
+      //seems far better for queries in the futur.
+      //but in that case, we should also keep the total
+      //number of events to be able to rebuild raw data
```

Comment:
Sometimes, we want the numbers to build new metrics, not only the percent, especially on aggreagation metrics.  
For ex, it allows to get the number of changes for a run. Or the number of non-compliance.